src/http/client/builder.rs

```diff
@@ -0,0 +1,585 @@
+use super::{HttpConnector, http_inspector::HttpVersionAdapter};
+use crate::{
+    Layer, Service,
+    dns::DnsResolver,
+    error::{BoxError, OpaqueError},
+    http::{Request, client::proxy::layer::HttpProxyConnector, dep::http_body},
+    net::client::{
+        EstablishedClientConnection,
+        pool::{
+            LruDropPool, PooledConnector,
+            http::{BasicHttpConId, BasicHttpConnIdentifier, HttpPooledConnectorConfig},
+        },
+    },
+    tcp::client::service::TcpConnector,
+};
+use std::{marker::PhantomData, time::Duration};
+
+#[cfg(feature = "boring")]
+use crate::tls::boring::client as boring_client;
+
+#[cfg(feature = "rustls")]
+use crate::tls::rustls::client as rustls_client;
+
+#[cfg(any(feature = "rustls", feature = "boring"))]
+use crate::http::client::http_inspector::HttpsAlpnModifier;
+
+#[cfg(feature = "socks5")]
+use crate::{http::client::proxy_connector::ProxyConnector, proxy::socks5::Socks5ProxyConnector};
+
+/// Builder that is designed to easily create a [`super::EasyHttpWebClient`] from most basic use cases
+#[derive(Default)]
+pub struct EasyHttpWebClientBuilder<C = (), S = ()> {
+    connector: C,
+    _phantom: PhantomData<S>,
+}
+
+pub struct TransportStage;
```

Comment:
Btw something I only notice now. But you usually want to make these marker types both: `non_exhaustive` and `Debug`.

Especially the first is important as we do not really want people to be able to construct this. What matters only is that the type itself is public.